libunwind/src/CompartmentInfo.hpp

```diff
@@ -13,21 +13,68 @@
 #ifndef __COMPARTMENT_INFO_HPP__
 #define __COMPARTMENT_INFO_HPP__
 
+extern "C" {
+
+#include <link.h>
+
+// Must mirror the layout in rtld_c18n_machdep.h
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(_LIBUNWIND_TARGET_AARCH64)
+struct dl_c18n_compart_state {
+  void *fp;
+  void *pc;
+  void *regs[10]; // c19 to c28
+  void *sp;
+};
+#endif
+
+#pragma weak dl_c18n_is_tramp
+int dl_c18n_is_tramp(ptraddr_t pc, void *tf) {
+  return 0;
+};
+
+#pragma weak dl_c18n_pop_trusted_stk
```

Comment:
These should probably be under #ifdef __CHERI_PURE_CAPABILITY__ so as not to mess with the symbol tables for non-CHERI unnecessarily. Technically one could guard them by AArch64 too but RISC-V will eventually want them so eh.